controller/konnect/konnectextension_controller.go

```
@@ -186,6 +187,9 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			if k8serrors.IsConflict(err) {
 				return ctrl.Result{Requeue: true}, nil
 			}
+			if k8serrors.IsNotFound(err) {
+				return ctrl.Result{}, nil
+			}
```

I don't think we want to silently ignore `NotFound` errors. These checks are after Update attempt, so `NotFound` means that something weird is happening, and proper error handling is required in my opinion.

---

This might happen for example when extension did not have the finalizer set yet and it got deleted between being queued for reconciliation and the `.Update()` call.

If the extension (or in general an object) is deleted, then there's no other action that we can do for it (apart from handling other resources that depend on this) IMHO.

WDYT?

---

Yes. When `Update()` returns `NotFound`, the object is already gone, so there's nothing more to do.